openpype/plugins/publish/integrate_subset_group.py

```diff
@@ -93,6 +93,6 @@ def get_profile_filter_criteria(self, instance):
         return {
             "families": anatomy_data["family"],
             "tasks": task.get("name"),
-            "hosts": anatomy_data["app"],
+            "hosts": anatomy_data.get("app"),
```

Comment:
Well if we change it then use the right key :)
```suggestion
            "hosts": instance.context.data["hostName"],
```